server/core/store.go

```diff
@@ -619,6 +626,77 @@ func (s *StoresInfo) SetRegionSize(storeID uint64, regionSize int64) {
 	}
 }
 
+const bytesPerMB = 1024 * 1024
+
+func calculateMaxScore(flexibleScore, capacity uint64) float64 {
+	return float64(capacity/bytesPerMB + flexibleScore)
+}
+
+func getMaxCapacity(stores []*StoreInfo) uint64 {
+	var maxCapacity uint64
+	for _, store := range stores {
+		capacity := store.GetCapacity()
+		if capacity > maxCapacity {
+			maxCapacity = capacity
+		}
+	}
+	return maxCapacity
+}
+
+func maxUint64(a, b uint64) uint64 {
+	if a > b {
+		return a
+	}
+	return b
+}
+
+func setAllMaxScore(stores []*StoreInfo, maxScore float64) []*StoreInfo {
+	results := make([]*StoreInfo, 0, len(stores))
+	for _, store := range stores {
+		results = append(results, store.Clone(SetMaxScore(maxScore)))
+	}
+	return results
+}
+
+// GetUpdatedMaxScoreStores update maxScore when flexible score change or add a new store
+func (s *StoresInfo) GetUpdatedMaxScoreStores(flexibleScore uint64, newStores ...*StoreInfo) []*StoreInfo {
+	results := make([]*StoreInfo, 0, len(newStores)+len(s.stores))
+	var newStore *StoreInfo
+	if len(newStores) != 0 {
+		newStore = newStores[0]
+	}
+
+	// add first store
+	if len(s.stores) == 0 {
+		if newStore != nil {
+			maxScore := calculateMaxScore(flexibleScore, newStore.GetCapacity())
+			results = append(results, newStore.Clone(SetMaxScore(maxScore)))
+		}
+		return results
+	}
+
+	// calculate max score
+	oldStores := s.GetStores()
+	oldMaxScore := oldStores[0].maxScore
+	maxScore := oldMaxScore
+
+	maxCapacity := getMaxCapacity(oldStores)
+	if calculateMaxScore(flexibleScore, maxCapacity) != oldMaxScore { //update by flexible score
```

Comment:
We currently only make the flexible score updating take effect when the new store is added. Do we need to make this configuration take effect dynamically? @nolouch @rleungx

Reply:
It seem to been supported without modifying